dev/com.ibm.ws.org.apache.aries.jndi.core/src/org/apache/aries/jndi/startup/Activator.java

```diff
@@ -0,0 +1,372 @@
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Supplier;
+
+import javax.naming.NamingException;
+import javax.naming.spi.InitialContextFactory;
+import javax.naming.spi.InitialContextFactoryBuilder;
+import javax.naming.spi.NamingManager;
+import javax.naming.spi.ObjectFactory;
+import javax.naming.spi.ObjectFactoryBuilder;
+
+import org.apache.aries.jndi.AugmenterInvokerImpl;
+import org.apache.aries.jndi.ContextManagerServiceFactory;
+import org.apache.aries.jndi.JREInitialContextFactoryBuilder;
+import org.apache.aries.jndi.OSGiInitialContextFactoryBuilder;
+import org.apache.aries.jndi.OSGiObjectFactoryBuilder;
+import org.apache.aries.jndi.ProviderAdminServiceFactory;
+import org.apache.aries.jndi.Utils;
+import org.apache.aries.jndi.spi.AugmenterInvoker;
+import org.apache.aries.jndi.tracker.CachingServiceTracker;
+import org.apache.aries.jndi.urls.URLObjectFactoryFinder;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleActivator;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.BundleEvent;
+import org.osgi.framework.Constants;
+import org.osgi.framework.ServiceReference;
+import org.osgi.framework.wiring.BundleRevision;
+import org.osgi.service.jndi.JNDIConstants;
+import org.osgi.service.jndi.JNDIContextManager;
+import org.osgi.service.jndi.JNDIProviderAdmin;
+import org.osgi.util.tracker.BundleTracker;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.ibm.ws.kernel.service.util.JavaInfo;
+
+/**
+ * The activator for this bundle makes sure the static classes in it are
+ * driven so they can do their magic stuff properly.
+ */
+public class Activator implements BundleActivator {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(Activator.class.getName());
+
+    private static final String DISABLE_BUILDER = "org.apache.aries.jndi.disable.builder";
+    private static final String FORCE_BUILDER = "org.apache.aries.jndi.force.builder";
+
+    private static volatile Activator instance;
+
+    private BundleTracker<ServiceCache> bundleServiceCaches;
+
+    private CachingServiceTracker<InitialContextFactoryBuilder> icfBuilders;
+    private CachingServiceTracker<URLObjectFactoryFinder> urlObjectFactoryFinders;
+    private CachingServiceTracker<InitialContextFactory> initialContextFactories;
+    private CachingServiceTracker<ObjectFactory> objectFactories;
+
+    private AugmenterInvoker augmenterInvoker;
+
+    private InitialContextFactoryBuilder originalICFBuilder;
+    private OSGiInitialContextFactoryBuilder icfBuilder;
+
+    private ObjectFactoryBuilder originalOFBuilder;
+    private OSGiObjectFactoryBuilder ofBuilder;
+
+    //In Java20+ the ObjectFactoryBuilder static field is in the NamingManagerHelper class.
+    private final Class<?> objectFactoryBuilderHolder = new Supplier<Class<?>>() {
+        @Override
+        public Class<?> get() {
+            if (JavaInfo.majorVersion() >= 20) {
```

Comment:
Instead of depending on Liberty JavaInfo I would just use code like this:

```java
        String version = System.getProperty("java.version"); // PUT in doPriv since this is static done in construction
        String[] versionElements = version.split("\\D"); // split on non-digits

        // Pre-JDK 9 the java.version is 1.MAJOR.MINOR
        // Post-JDK 9 the java.version is MAJOR.MINOR
        int i = Integer.valueOf(versionElements[0]) == 1 ? 1 : 0;
        int majorVersion = Integer.valueOf(versionElements[i++]);
```

Reply:
Removed use of JavaInfo